### ISSUE TITLE
Ultimate 64: A little glitch

### ISSUE THREAD

**Issue by mbwum73** [GideonZ/1541ultimate#116]
The very simple demo called "Rampage Demo" has a tiny glitch in the second part of the demo.
The demo itself can be found here: http://www.vdpoel.org/RampageDemo.d64

When I run this demo on my Ultimate 64 (firmware 1.21) it starts just fine. However when you press space bar the second part of the demo begins. In this part you see some monsters with a great graphic trick. However this is not the problem. Above the monsters is a scrolling text. When you look at the rigt, just below this text, you can see a small flickering line. This line is not present on a real C64 and also not on any emulator I've tried it on. So it seems to be something that only happens on the Ultimate 64. Although it's a minor glitch, I hope it can be fixed.

**Comment by mbwum73** [comment]
Just tried it with the new 1.24 firmware. The glitch is still there unfortunately.

**Comment by GideonZ** [comment]
Of course.. The release notes do not mention that I have worked on the C64
emulation in any way. :-)

**Comment by mbwum73** [comment]
I know. Was simply trying it on the new firmware. Off course the new firmware is great and all your hard work is very much appreciated.  So I hope I didn't offend you by reporting the glitch. It's very minor but I just love the Ultimate 64 and hope it will get as close to a real C64 as possible. All new features are great though.

**Comment by GideonZ** [comment]
It is totally fine. I just don't think it is necessary to reiterate that
the issue has not yet been fixed. It is just like what I tell my wife, if
she asks me to fix something in the house: "you don't need to remind me
every six months.." Haha.

**Comment by Jumpman64** [comment]
omg, that's a tiny lil' bugger to find. being curious i tried it too. found the demo elsewhere, made a screenshot and attached both for future convenience and reference.  (this could apply to our wives too :D )

![rampage_demo_pixel_glitch](https://user-images.githubusercontent.com/50739888/81119873-ec5b6600-8f2b-11ea-993b-4f26012f738f.png)
[rampagedemo.zip](https://github.com/GideonZ/1541ultimate/files/4583804/rampagedemo.zip)

**Comment by mbwum73** [comment]
Thank you for trying it out and making the screenshot of it.
I know it' a very minor glitch, but on a real C64 (and Vice, Hoxs etc) it's not there. So it's defenitly something that's not entirely accurate in the Ultimate 64. It's the only glitch I've fond so far, so it's probably a hard one to fix, because it can't be reproduced with another demo or game.

________________________________
Van: Jumpman64 <notifications@github.com>
Verzonden: woensdag 6 mei 2020 00:01
Aan: GideonZ/1541ultimate <1541ultimate@noreply.github.com>
CC: mbwum73 <renepela@hotmail.com>; Author <author@noreply.github.com>
Onderwerp: Re: [GideonZ/1541ultimate] Ultimate 64: A little glitch (#116)


omg, that's a tiny lil' bugger to find. being curious i tried it too. found the demo elsewhere, made a screenshot and attached both for future convenience and reference. (this could apply to our wives too :D )

[rampage_demo_pixel_glitch]<https://user-images.githubusercontent.com/50739888/81119873-ec5b6600-8f2b-11ea-993b-4f26012f738f.png>
rampagedemo.zip<https://github.com/GideonZ/1541ultimate/files/4583804/rampagedemo.zip>

**Comment by Jumpman64** [comment]
n/p, we are community. :) It's "just an inaccuracy" that is still open. There are many test cases used for testing on emulators n stuff. Here's a nice site where the tests get documented: https://www.c64-wiki.de/wiki/Ultimate_64/Testbericht
Sadly my assembler knowledge is lousy otherwise i would dig into that and see what techniques are used in this part of the demo. I guess Gideon is fixing the remaining things one by one and we'll get a perfect C64 in fpga in the future. Would be legendary.

**Comment by t-h-e-joker** [comment]
I have tested this demo with many different, real C64 boards. The following findings were made:

On every PAL-board Assy.250469 (VIC-II = 8565) the glitch does not occur.
On every other PAL board Assy. 250407/250425/250466 (VIC-II = **6569**) **the glitch occurs**.

Conclusion: With this demo you can see very well if an 8565 or a 6569 is used in your real C64, but it is not a U64 issue if the 6569 is emulated, which is shown in the Giana-Sister intro.

**Comment by Jumpman64** [comment]
@GideonZ You are not going to implemet a VIC-II selector, do you? ; )
t-h-e-joker solved this case, maybe you want to close this issue...
